registry/handlers/app.go

```diff
@@ -411,3 +381,38 @@ func apiBase(w http.ResponseWriter, r *http.Request) {
 
 	fmt.Fprint(w, emptyJSON)
 }
+
+func accumulateAccessRecords(method string, repo string, records []auth.Access) []auth.Access {
```

Comment:
This should be called `appendAccessRecords`, with `records` being the first argument, since it is the destination.